core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -729,16 +743,16 @@ public static Builder builder(MetaData metaData) {
     }
 
     /** All known byte-sized cluster settings. */
-    public static final Set<String> CLUSTER_BYTES_SIZE_SETTINGS = ImmutableSet.of(
+    public static final Set<String> CLUSTER_BYTES_SIZE_SETTINGS = unmodifiableSet(newHashSet(
```

Comment:
I know this is kind of a burried place to have the discussion, but what is wrong with `newHashSet` as a utility? That people will use it and not see the implicit array creation?